src/components/probe/index.ts

```diff
@@ -222,3 +231,20 @@ export async function doProbe(
     }
   }
 }
+
+function calculateResponseTime(probeRes: ProbeRequestResponse) {
+  responseCount += 1
+
+  if (responseCount === 1) {
+    // first time
+    maxResponseTime = probeRes.responseTime
+    minResponseTime = probeRes.responseTime
+  } else if (probeRes.responseTime > maxResponseTime) {
+    maxResponseTime = probeRes.responseTime
+  } else if (probeRes.responseTime < minResponseTime) {
+    minResponseTime = probeRes.responseTime
+  }
+
+  totalResponseTime += probeRes.responseTime
+  averageResponseTime = Math.floor(totalResponseTime / responseCount)
+}
```

Comment:
1.  can we separate this into different file for the sake of separation of concern?
2. It should calculate max, min, and ave in the last 24 hours. Maybe add another variable to store a starting timestamp, then reset the max,min,and ave values in this function when current - starting is 24 hours.